pages/index.js

```diff
@@ -1,16 +1,100 @@
-import FooterButton from "../src/components/Button/FooterButton";
-import Days from "../src/components/Days/Days";
+import Days from "../src/components/Days";
+import Link from "next/link";
+import useStore from "../src/store";
+import { useState, useEffect } from "react";
 
 export default function HomePage() {
+  const { days, toggleDay, sessions } = useStore();
+  const addedDays = days.filter((day) => day.added);
+  const [selectedType, setSelectedType] = useState("short");
+
+  useEffect(() => {
+    generateSessionsForDays(addedDays, sessions);
+  });
+
+  function generateSessionsForDays(days) {
+    days.forEach((day) => {
+      if (!day.added) {
+        day.sessions = [];
+      } else {
+        const remainingSessions = sessions.filter(
+          (session) =>
+            session.eventDistance === selectedType &&
+            session.type !== day.type &&
+            !day.sessions.includes(session)
+        );
+        const selectedSessions = [];
+        for (let i = 0; i < 2; i++) {
+          const randomIndex = Math.floor(
+            Math.random() * remainingSessions.length
+          );
+          const session = remainingSessions[randomIndex];
+          selectedSessions.push(session);
+          remainingSessions.splice(randomIndex, 1);
+        }
+        day.sessions = selectedSessions;
+      }
+    });
+  }
+
+  function handleCreatePlanClick() {
+    generateSessionsForDays(addedDays);
+  }
+
+  function handleDayToggle(day) {
+    toggleDay(day.id);
+    generateSessionsForDays(addedDays);
+  }
+
   return (
     <div>
+      <h2>Choose your event distance</h2>
+      <form>
+        <label>
+          Short Distance Triathlon
+          <button
+            type="button"
+            className={selectedType === "short" ? "selected" : ""}
+            onClick={() => setSelectedType("short")}
+          >
+            {selectedType === "short" ? "Selected" : "Select"}
+          </button>
+        </label>
+        <br />
```

Comment:
Maybe write yourself a reminder to remove the <br> tags when starting with the styling